opennms-webapp/src/main/webapp/includes/grafana-box.jsp

```diff
@@ -128,9 +136,14 @@
                     }
                 }
                 if (showDashboard) {
-                    $('#dashboardlist').append('<a href="<%=grafanaProtocol%>://<%=grafanaHostname%>:<%=grafanaPort%>/dashboard/' + val['uri'] + '"><span class="glyphicon glyphicon-signal" aria-hidden="true"></span>&nbsp;' + val['title'] + "</a><br/>");
+                    if (limit < 1 || count++ < limit) {
+                        $('#dashboardlist').append('<a href="<%=grafanaProtocol%>://<%=grafanaHostname%>:<%=grafanaPort%>/dashboard/' + val['uri'] + '"><span class="glyphicon glyphicon-signal" aria-hidden="true"></span>&nbsp;' + val['title'] + "</a><br/>");
+                    }
                 }
             };
+            if (limit > 0 && count > limit) {
```

Comment:
No, it's correct since count is increased in the preceding if-statement. So, count will match limit in the case limit equals the number of available dashboards. In this case we do not need the "View list of all Dashboards" link.